src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseWrapper.cs

```diff
@@ -238,57 +223,67 @@ private static MongoUpdate ConvertModifiedEntryToMongoUpdate(string collectionNa
-        var idValue = entry.GetCurrentValue(idProperty);
-        var idSerializationInfo = GetPropertySerializationInfo(entitySerializer, idProperty);
-        var idSerializer = idSerializationInfo.Serializer;
-        var serializedIdValue = SerializeValue(idSerializer, idValue);
-        var filterDefinition = Builders<BsonDocument>.Filter.Eq("_id", serializedIdValue);
-
         var fieldValues = new BsonDocument();
         foreach (var property in entityType.GetProperties())
         {
             if (entry.IsModified(property))
             {
                 var propertyValue = entry.GetCurrentValue(property);
                 var propertySerializationInfo = GetPropertySerializationInfo(entitySerializer, property);
-                var elementName = propertySerializationInfo.ElementName;
                 var propertySerializer = propertySerializationInfo.Serializer;
                 var serializedPropertyValue = SerializeValue(propertySerializer, propertyValue);
+
+                var elementName = propertySerializationInfo.ElementName;
                 fieldValues[elementName] = serializedPropertyValue;
             }
         }
+
         var updateDocument = new BsonDocument("$set", fieldValues);
         var updateDefinition = new BsonDocumentUpdateDefinition<BsonDocument>(updateDocument);
 
-        var model = new UpdateOneModel<BsonDocument>(filterDefinition, updateDefinition);
+        var idFilter = CreateIdFilter(entitySerializer, entry);
+        var model = new UpdateOneModel<BsonDocument>(idFilter, updateDefinition);
         return new MongoUpdate(collectionName, model);
+    }
 
-        static BsonSerializationInfo GetPropertySerializationInfo(IBsonDocumentSerializer entitySerializer, IReadOnlyProperty property)
-        {
-            if (entitySerializer.TryGetMemberSerializationInfo(property.Name, out var serializationInfo))
-            {
-                return serializationInfo;
-            }
+    private static FilterDefinition<BsonDocument> CreateIdFilter(IBsonDocumentSerializer serializer, IUpdateEntry entry)
+    {
+        var idProperty = entry.EntityType.GetIdProperty() ??
+                         throw new InvalidOperationException($"Type {entry.EntityType.ClrType.Name} has no Id property.");
 
-            throw new InvalidOperationException($"Unable to get serialization info for property: {property.Name}.");
-        }
+        var idSerializationInfo = GetPropertySerializationInfo(serializer, idProperty);
+
+        var idValue = entry.GetCurrentValue(idProperty);
+        var serializedIdValue = SerializeValue(idSerializationInfo.Serializer, idValue);
 
-        static BsonValue SerializeValue(IBsonSerializer serializer, object? value)
+        return Builders<BsonDocument>.Filter.Eq("_id", serializedIdValue);
+    }
+
+    private static BsonSerializationInfo GetPropertySerializationInfo(IBsonDocumentSerializer entitySerializer,
+        IReadOnlyProperty property)
```

Comment:
As a readability issue I prefer all parameters on a single line, or once you need multiple lines, one parameter per line.

At a quick glance this declaration "looks" like it only takes a single parameter called `property`.

    private static BsonSerializationInfo GetPropertySerializationInfo(
        IBsonDocumentSerializer entitySerializer,
        IReadOnlyProperty property)

Reply:
Yeah I agree with you here, this is how Rider is formatting the file using the .editorconfig we have - I need to see if we can tweak the settings to see if we can get it to do this for us.